### PR TITLE
Allow configuration checks to be turned off

### PR DESCRIPTION
**Pull request by coreone**
I was having the same issue as described in #164 so I added this so the checks could be turned off if needed.

### CODE REVIEW

File: spec/classes/dns_init_spec.rb [hunk 1]
```diff
@@ -365,6 +365,21 @@
         it { should contain_file("#{etc_directory}/dns-key.key") }
       end
 
+      describe 'with config_check set to false' do
+        let :params do
```
Comment (alexjfisher) [inline spec/classes/dns_init_spec.rb:369]: For consistency with the other tests, maybe
`let(:params) { {:config_check => false} }`??

File: spec/classes/dns_init_spec.rb [hunk 2]
```diff
@@ -365,6 +365,21 @@
...
+          {
+            'config_check' => false,
+          }
+        end
+
+        it { is_expected.to compile.with_all_deps }
+
+        it {
+          is_expected.to contain_concat("#{etc_directory}/named.conf")
+            .with_validate_cmd(nil)
```
Comment (alexjfisher) [inline spec/classes/dns_init_spec.rb:379]: Consider `.without_validate_cmd`

**Review by alexjfisher — APPROVED** [review]
LGTM other than a couple of very minor style suggestions.

File: manifests/init.pp [hunk 3]
```diff
@@ -168,6 +168,7 @@
   Variant[Enum['running', 'stopped'], Boolean] $service_ensure      = $dns::params::service_ensure,
   Boolean $service_enable                                           = $dns::params::service_enable,
   Optional[String[1]] $service_restart_command                      = $dns::params::service_restart_command,
+  Boolean $config_check                                             = $dns::params::config_check,
```
Comment (alexjfisher) [inline manifests/init.pp:171]: You need to add a puppet strings style documentation comment.

**Review by alexjfisher — CHANGES_REQUESTED** [review]

**Review by alexjfisher — APPROVED** [review]

**Review by ekohl — APPROVED** [review]

**Pull request merged**